### DISCUSSION TITLE
pre-request scripts

### DISCUSSION THREAD

**Discussion by scollovati** [Kong/insomnia#3806] [Ideas] [answered]
What about implementing a feature like this one https://learning.postman.com/docs/writing-scripts/pre-request-scripts/ ?

**Comment by dimitropoulos** [comment]
Could you please provide a use-case for what you'd use this for?  What's something you'd like to do today with insomnia that you can't without a feature like this?

**Reply by mbunderline76** [reply]
in our case before every request we must send a request to get token!
without this feature I cant use insomnia

**Reply by sebastianbuechler** [reply]
Format the json body to have no space and no new lines in order to not get blocked by a WAF rule.

**Reply by zillou** [reply]
Calculating the request digest and signature is a quite common thing for API Key and API Secret based authentication.

**Reply by JuninhoFreitas** [reply]
I need to generate a payload that uses random uuidv4, names unically based in timestamp and other random datas,
then, i use the data created in environment to utilize others endpoints,
So, basically, what i needed was:
Send a request,
Use some parts of response in environment,
Use this environment to utilize another endpoin

**Reply by opctim** [reply]
Automatically perform a JWT login, fetch the token and attach it to the request body if not present or expired.

**Reply by lcsmobileqa** [reply]
I need to pre-format a date to use as the variable in multiple of my requests. This was a hugely useful feature in Postman and would be most welcome here. Being able to set the date/time format allowed our tests to automate hands-off from gitlab. Without it, we're having to manually update before running the collection. Thank you!

**Reply by morfidon** [reply]
I could stringify json for next api call that requires it stringified for example

**Comment by peronecode** [comment]
Dammit, here I am begging for this feature hahah.
It will be awesome to have this feature. What do you think @gschier?

If needs help to develop, I'm not into this code but I'm in. 🤘🏼

**Reply by dimitropoulos** [reply]
What is your use-case?

**Reply by peronecode** [reply]
Sorry my late answer peps, but I'm looking for a pre-request script execution, like we have in Postman. 🤓 

Use-case: _Before every single request, we can add a JavaScript (or any language) small logic script that suppose to run before the request._

e.g: _Before my request, I would love to add a new custom hash generated to a new variable, and this is a JavaScript logic. So, after ran this script/code block/logic, my request will be sent using the updated variable that was updated over the pre-request script._

**Comment by Vagrantin** [comment]
Hello,
I’m also interested in this feature.
In my case I’m using pre-request script to populate variable before triggering the API call.
This is use full when triggering the API call several time and updating the variables before each call.
And my API call is more clean when there is lot’s of data in it.
Hope that helps.

**Comment by steelerc69** [comment]
My use case is to calculate a HMAC value from the body provided, before I send a request and populate a header value with it

**Comment by jquexxx** [comment]
hi chiming in here, my use-case is to insert unwanted characters during api call. 

thanks.

**Comment by felipejsm** [comment]
Hi,
My use case is slightly different from the others
I use it to do a http request to get a token and use it as a header on other http request

**Reply by GleytonLima** [reply]
We need also a custom script after the request to extract and compare some fields to populate environment variables. Like the Test tab of Postman, for example.

**Comment by T2MWorks-Mitch** [comment]
sadly I will have to go back to postman without this feature in insomnia,  as the  postman API collections I import/use , leverage this feature for ading API keys to headers prior to each request 

as a non developer i jsut dont have the bandwidth to figure out how to hack this together in Insomnia

**Comment by GleytonLima** [comment]
I made a solution using some custom plugins to get the same pre and post request handling functionality. I made a video, in portuguese, explaining how to use these features (https://youtu.be/NO8p_4vK1G4?t=591). But a pre and post request area built-in would be ideal.

**Comment by lookto** [comment]
This would be such a good feature. In my use-case i need to get an acess token for the route i'm trying to test.

**Comment by brpeterman** [comment]
I have the same use case. In Postman, I was using the pre-request script to cache an auth token in the environment, then populating the Authorization header from that environment variable. Whenever the token expired, the script would automatically retrieve a new one.

**Comment by naretini** [comment]
This solves your case : https://docs.insomnia.rest/insomnia/chaining-requests
Chain token request response to the final request  Auth value.
Postman uses the pre-script concept, Insomnia has this chain requests, which is not identical, maybe limiting, but for the purpose of auth works.

**Reply by brpeterman** [reply]
Thank you! I'll give that a try.

**Reply by brpeterman** [reply]
Following up on this, request chaining is working really well already. I've placed the tag in an environment variable and each of my requests is pulling that variable in for the bearer token. Thanks for the tip!

**Reply by T2MWorks-Mitch** [reply]
@naretini - this does not actaully solve the case - I use this now for obtaining a token , but still have to run the acquire token command first. Postman has this functionalilty AND the ability to run the "acquire token" request automatically as part of the second request

**Comment by henrycorredor** [comment]
In our case, we create a sign with the CryptoJS library with a given string, extracted from some keys of the body. It seems that sign can be only made on the go when you make the post request.

**Comment by manhtranlinh** [comment]
In my case, I need to send JSON message and some attachments  in one POST request by multipart-data/form option. But in case option is multipart-data/form, Insomnia will set JSON part as string and send to server, so that I have to decode this part. So that if has pre-request scripts or option to combine STRUCTURED/Multipart Form and TEXT/JSON is very comfortable.

**Comment by KikoCosmetics** [comment]
Hi!

Any news on this? I actually want to migrate from Postman, but I used pre-request scripts to generate headers for requests to AWS.
Without this I can't really move to Insomnia!

**Comment by DanielHilton** [comment]
I would also be very interested to have this feature.

My usecase is:
I need to generate a SHA256 checksum of several concatenated values from a request body along with a secret key. In Postman I can do this in a pre-request script, in Insomnia I cannot and must do it manually.

**Comment by mjr2595** [comment]
Just keeping this thing alive! I would love to be able to fully use Insomnia, but an API I need for work requires the pre-request functionality to construct the signature. Pretty much the same use case as @steelerc69

**Comment by torian257x** [comment]
we need prerequest as well to generate hmac tokens. Sadly

e.g. add the prerequest script feature please

**Comment by Bromles** [comment]
Today I needed to remove 1500 issues that were mistakenly imported from Jira to Gitlab. There was no mass deletion in Gitlab itself, but there is an API for deleting an issue by its id

As a result, the task was reduced to sorting out all the identifiers of all the issues, and for each send a request to the Gitlab API for deletion

In Postman, with the help of pre-request scripts and a collection of 1 request, all this was completely automated, including obtaining a token. All 1500 issues were deleted in about 5 minutes, given the artificial delay of 50ms between requests

In current Insomnia, this would be much more difficult to automate, if not impossible. Please add this feature

**Reply by iulianraduat** [reply]
My use case is to set a variable to a random number (or guid) and us the value of this variable in the body of a POST call. This is part of testing if the creation part of CRUD API works.

Is it there the possibility to create a plugin which add the "Pre-request script" as an additional tab to the default JOSN/Auth/Query/Headers/Docs?

Thanks,
Iulian

**Comment by torian257x** [comment]
guys, the answer is already out. It is literally:
write your own plugin. Sounds scary I know. But they made it really easy

its literally chrome running insomnia
![image](https://user-images.githubusercontent.com/2072058/221265865-2421ebe4-4d7d-4fa8-9678-6a5d202d2ce4.png)

there is a button to reload changes
![image](https://user-images.githubusercontent.com/2072058/221266026-c70d294f-c9b5-4ba0-af58-e6a61abaa66d.png)

all you need is NPM. Writing a plugin literally is something like https://github.com/anson-vandoren/insomnia-plugin-binance-signing/blob/master/insomnia_plugin_binance_signing.js

and beware, 90% of his code is "if request is already signed, bail" 

Follow that guys plugin stuff, its super simple. We dont need a pre request script thingy. Plugins do exactly that.

https://ansonvandoren.com/posts/hmac-signing-insomnia/

**Comment by iulianraduat** [comment]
@torian257x I accepted your challenge and I created an Insomnia plugin called "[Custom Generated Value](https://www.npmjs.com/package/insomnia-plugin-custom-generated-value)".

The solution was to define a new TemplateTag called "Custom Generated Value" which can be used in more than just request's body (try to insert it using ctrl+space to display the list of available tags).

The code for generated value is defined either directly in an environment variable (as an array of lines) or in a .js file (which has its name in the environment variable). You can see some examples in the [README](https://www.npmjs.com/package/insomnia-plugin-custom-generated-value) of this plugin.

Also it makes all defined environment variables available to your code via variable $$ so you can use them to build your result.

You can find it in the [Insomnia's Plugin Hub](https://insomnia.rest/plugins) at https://insomnia.rest/plugins/insomnia-plugin-custom-generated-value.

Please try it and let me know what you think.

Thanks,
Iulian

**Comment by vlcekr** [comment]
Hi, are there any plans to implement this feature please? Like for others, for me this is crucial feature too.

**Comment by JuninhoFreitas** [comment]
**My Workaround:**
Run a local api just to in response return the JSON body sent,
then, in environment, refer to the response of the mock api, 
and create a folder with multiple cases that u will need, like, "pre-req auxiliar", "pre-req master", "pre-req x".

Code:
https://gist.github.com/JuninhoFreitas/d7a6ef58c45149d62dd0025951d96bc1

Only needs nodejs, to keep running in background, use pm2
e.g:
```pm2 run index.js```


**My case:**
I need to generate a payload that uses random uuidv4, names unically based in timestamp and other random datas, 
then, i use the data created in environment to utilize others endpoints,
So, basically, what i needed was:
Send a request, 
Use some parts of response in environment,
Use this environment to utilize another endpoint.

**Comment by sebastianbuechler** [comment]
To be honest, this was the reason why we switched to postman. No need for workarounds or custom solutions. It's just there and working.

**Reply by subnetmarco** [reply]
Many changes are coming to Insomnia and as soon as we implement SSE (currently being worked on) I am eying this capability next.

**Reply by sebastianbuechler** [reply]
Thanks for the response @subnetmarco. Just wanted to give some information about why we switched. I still like the approach of Isnomnia, but that was a deal-breaker. Maybe we'll evaluate it in the future again.

**Reply by hbertsch** [reply]
We also are using `Postman` for the same reason. However they will move their product entirely to the cloud. We will not be allowed to use this product... Its a shame

**Comment by NickRimmer** [comment]
The most challenging part will be not creating a plugin with input for JS and even not about how to run it before request. But what should this script be for? Changing of variables in request? Replace/Inject body? Build/Update the request URL? I think the most difficult part will be to add access for a script to Insomnia world, as Insomnia provided very limited access to application variables/methods even for plugins

**Comment by hbertsch** [comment]
The **pre-request feature** is a critical functionality that we would require in `Insomnia`, similar to what is available in `Postman`. This feature is one of the major advantages of Postman, and we heavily rely on it in our collections. However, since Postman has transitioned to a full cloud-based platform, we are no longer able to use it due to compliance and security concerns. It is not feasible for us to store our company's API details and scripts on a third-party cloud server.

As a result, we are actively searching for an alternative tool, and `Insomnia` appears to be a promising option. However, the absence of the pre-request feature in Insomnia is a significant drawback for us and could potentially hinder us from using it. This feature is essential because we have a variety of logic that must be executed before and after certain requests, such as setting environment variables, hashing data, and generating payloads. Without this capability, our testing process would be severely impacted.

**Reply by iulianraduat** [reply]
Friendly reminder: 
I had a similar problem and hence I created an Insomnia plugin called "[Custom Generated Value](https://www.npmjs.com/package/insomnia-plugin-custom-generated-value)". Until an official solution is provided, I think that this plugin can solve a lot of use cases. You can even make feature requests if you need something added to this [plugin](https://www.npmjs.com/package/insomnia-plugin-custom-generated-value).

You can find more details in my post from Feb 25

Thanks,
Iulian

**Reply by JPPereira123** [reply]
Same here..  We're currently doing a proof-of-value on Postman vs Insomnia enterprise options for hundreds of seats, and even though Insomnia overall looks quite promising, not having similar pre/post-req script capabilities seems like quite a deal breaker for us.

**Comment by subnetmarco** [comment]
Update: we are currently allocating resources to work on this issue, we will post more updates once we have a better idea on the timeline.

**Reply by JuninhoFreitas** [reply]
God Heard the People!
Thanks Marco!

**Reply by DanielHilton** [reply]
many thanks @subnetmarco i looks forward to hearing progress 🙏🏻

**Reply by JPPereira123** [reply]
Thank you Marco.

Just a heads up so you don't run into the same experience we have with Postman: approving pre/post request script PRs via github is very bad UX as Postman formats the scripts in a very text/json unreadable manner. I'd recommend investigating ways to execute plain js/ts files so they can be stored and reviewed as usual.

**Reply by caseyfw** [reply]
I concur with this - it would be ideal if there was a mechanism to have these scripts live in actual JS files rather than JS embedded in JSON like Postman's solution.

All of our PRs to workflow updates look like this, and they're impossible to lint/review/edit/etc:
<img width="1195" alt="image" src="https://github.com/Kong/insomnia/assets/248665/50ee3f9e-d8bd-4c37-a875-4607324c7495">

**Reply by subnetmarco** [reply]
Makes sense, easier to debug when the scripts are not embedded elsewhere. We will see what we can do (+ @ihexxa ).

**Comment by subnetmarco** [comment]
**Update**: we are making progress here, we may push this capability in beta in early February.

**Bonus point**: we have introduced a compatibility layer with Postman's SDK, so the same code can run in Insomnia with little or no modification.

<img width="1725" alt="Screenshot 2024-01-02 at 7 46 39 AM" src="https://github.com/Kong/insomnia/assets/964813/bfef1870-bb14-45aa-a053-07483a80be08">

**Reply by khirasoft** [reply]
Cool! Where can I sign in for the beta?
Also one additional question: Will it be possible to include other libs like crypto-js for example?

**Reply by guddu2112** [reply]
Thanks !!! We are also looking for this feature eagerly. let us know if there is a beta available already for observation feedback.

**Reply by subnetmarco** [reply]
I will be sharing a video soon, we may enable it in beta this month, this feature is quite large so there is a big ongoing review as we speak.

**Comment by subnetmarco** [comment]
Here is a preview video of pre-request scripting that we are building:

**📹 https://www.loom.com/share/9e6f1253f8ca48bca49788f94fde8063**

You can see how to:

1. Create a script inside Insomnia.
2. Manipulate environments from within a script.
3. Send requests and access responses.
4. Require modules for more advanced use-cases.
5. Logging for debugging and more.
6. **Compatibility with no modifications with Postman scripts.** by providing a `pm.*` alias that simply points to `insomnia.*` and implements the same interfaces (!). This will simplify migration from Postman.

We want to ship a preview of this capability ideally this month.

**Reply by guddu2112** [reply]
Hi Any update on the new version availability supporting this feature?

**Reply by subnetmarco** [reply]
We are planning to ship a beta closer by end of month, with GA sometime next month.

**Reply by ShakedAharonn** [reply]
Hi, is the beta already out?
Is it possible to join it? and if not any estimation to when the GA will be available?

**Reply by subnetmarco** [reply]
Hello, I will share a link to a beta distribution later this week. This capability ships this month.

**Reply by TomaszPanek** [reply]
Any updates about this @subnetmarco  ?

**Reply by subnetmarco** [reply]
@TomaszPanek see the latest update on this discussion.

**Comment by subnetmarco** [comment]
**The beta release with pre-request scripting is out!**

You can download it at https://github.com/Kong/insomnia/releases/tag/core%409.0.0-beta.1

With this new release you will see a new "Pre-request Script" tab in the collection view, where you can create scripts:

* This is a **beta feature** and we are looking for feedback for improvements. Please send your feedback so we can incorporate it before GA: bugs, improvements or missing compatibility with Postman.
* The `insomnia.` object can also be consumed with `pm.*`, they are equivalent.
* The goal is to fully provide Postman compatibility for most of their API, so feel free to try and let us know if there is an API that is missing in Insomnia. We know a couple are missing, but we first want to see if the community cares about those.

<img width="871" alt="Screenshot 2024-03-09 at 8 58 37 AM" src="https://github.com/Kong/insomnia/assets/964813/292a242c-19f3-484e-af9e-c4d50d138f9b">

**Reply by deepanshuds** [reply]
Thanks for releasing this as a Beta version! 
I tried importing my collection from Postman which uses the random function in Pre-request script. Insomnia does not identify the "_" used in the command and throws an error when the request is run. 

<img width="514" alt="image" src="https://github.com/Kong/insomnia/assets/55419756/7376ba13-7191-4d69-8864-2772a2f162ac">

Could you please check if this can be added to this release to improve compatibility? Thanks.

**Reply by subnetmarco** [reply]
@deepanshuds yep, we will check it. Thanks for the feedback.

**Reply by ihexxa** [reply]
The issue is fixed in #7172 and theoretically will be included in the next release.

**Reply by AleksandrZHANG** [reply]
We need Base64 key generation using CryptoJS while it seems not defined

**Reply by ihexxa** [reply]
Hi @AleksandrZHANG, this is expected as it is also  [a part of the above change](https://github.com/Kong/insomnia/pull/7172/files#diff-9e680b2f1ce3058f0649cc4de021ccdee89782c1f3a91f819e0352f5aa6f5de8R57) and in review.

**Reply by ihexxa** [reply]
@deepanshuds @AleksandrZHANG In case you missed it, external modules are available in the latest beta4 version: https://github.com/Kong/insomnia/releases/tag/core%409.0.0-beta.4

**Reply by rickseven** [reply]
Hi @ihexxa, I have installed latest beta4 version, I need "crypto" module in my case but it seems not defined.
![image](https://github.com/Kong/insomnia/assets/22319137/ba7686f5-f612-469d-a3d5-b89bee430953)

**Comment by subnetmarco** [comment]
A new beta version has been released with notable improvements to the pre-request scripting capability: https://github.com/Kong/insomnia/releases/tag/core%409.0.0-beta.3

We have another couple of known improvements that we want to make, before releasing it as GA in the next few weeks.

**Reply by lmeyer92** [reply]
Looks good. Is there a way to share scripts among multiple requests though ? Also, from my testing, Insomnia cannot import scripts from Postman's export when importing a collection.

**Reply by subnetmarco** [reply]
> Is there a way to share scripts among multiple requests though

We will build this, but now we are focused on shipping the baseline capability first.

> Also, from my testing, Insomnia cannot import scripts from Postman's export when importing a collection.

Using the latest v9.0-beta3? cc @ihexxa

**Reply by rickseven** [reply]
Hi @subnetmarco I have installed the beta.3 version

I tried to write code in the pre-request script which is just a console log like this
![image](https://github.com/Kong/insomnia/assets/22319137/b6d8fac5-6c40-4491-b2ff-3102915d6c11)

When I try to send the request, I get an error like this
![image](https://github.com/Kong/insomnia/assets/22319137/36d97376-f0b4-46fc-9bef-56e8741b1588)

But when I delete the code from the pre-request script, the request is sent successfully.

**Reply by subnetmarco** [reply]
Thanks for the feedback, we will get this fixed.

**Reply by rickseven** [reply]
Thank you, please let me know if it has been fixed.

**Reply by ihexxa** [reply]
Hi @rickseven, could you try to remove 1) envs or variables in URL or query parameters 2) variables in auth methods 3) variables in headers 4) variables in body, and see if it still happens? Or it would be great if you could share more steps for reproducing, really appreciate.

**Reply by rickseven** [reply]
Hi @ihexxa , it seems I have found the cause, if I change the base url using the env variable the error will appear, but if I don't use the env variable and write the full url directly it will work. 

Please see the following recording to reproduce it.
https://jam.dev/c/b4cebbf0-0c9d-4c9a-a18b-2b909d2e5157

And this is the env that I use.
![image](https://github.com/Kong/insomnia/assets/22319137/3e26839d-43b0-4230-976d-c6e0c6779067)

**Reply by ihexxa** [reply]
PR #7249 is pushed to fix the "unexpected token" problem.

**Reply by ihexxa** [reply]
Hi @lmeyer92, are you using the v9.0-beta3? If it still happens, would you mind to share a minimal export which can help to reproduce? Thanks a lot.

**Reply by rickseven** [reply]
Thank you @ihexxa,

I'll try again when it's released

**Reply by rickseven** [reply]
Hi @ihexxa  it looks like we need to include at least one reviewer for this PR
![image](https://github.com/Kong/insomnia/assets/22319137/7c17ec53-3b89-4775-8779-9c4faeb2b6b6)

**Reply by solorisx** [reply]
Hi, are you planning to extend the pre-request script functionalities to Unit Testing? It would be very useful to get / set variables.

**Reply by ihexxa** [reply]
Hi @solorisx , the post-request script will be started right after the pre-request script release, would it cover your requirement?

**Reply by solorisx** [reply]
Hi @ihexxa
(deleted my earlier comment - did not mean to create a top level comment)
By post-request scripts, you mean what Postman calls "Tests"?

That would certainly help - but what I meant was the ability to do
something in a test like:

insomnia.set_variable("variable_name", value);
const response = await insomnia.send();
// assertions here

This would save us from having to create dedicated requests for each test
where we want to verify the response. I think issue [#3218](https://github.com/Kong/insomnia/issues/3218) talks about the
same thing.

**Reply by misaya98** [reply]
for enterprise user,  when we can get the pre-requst-script feature ?

**Reply by ihexxa** [reply]
@misaya98 It will be released in this month, and one or more beta version(s) will be released before that.

**Reply by deepanshuds** [reply]
Hi @ihexxa,
I tried to import a request from POSTMAN which has variables defined in pre-script and it failed to run in Insomnia. It throws an error 'The request failed due to an unhandled error: Error: attempted to output null or undefined value'. 

var tempVar = "test" + _.random(1,1000000);
pm.environment.set("tempVar", tempVar);
var id = "ABC" + tempVar 
pm.environment.set("id", id);

The issue is with using variables, it works fine if I hardcode the value of the variable. 

Also, if I do not define the same variables as part of my environment variables and try to use them as part of the request body, it is unable to identify the variable. Would it be possible for the body to read the variables defined in pre-request scripts once it's defined there?
![image](https://github.com/Kong/insomnia/assets/55419756/c48c61df-14e4-4048-b53b-4558471ee3b0)

**Reply by ihexxa** [reply]
Hi @deepanshuds , thanks for the feedback. Ah I'm not able to reproduce it, would you mind to share one minimal export for reproducing it? Please check if all variables in url, body, auth, headers are defined. Probably you could also try `conole.log` to debug it.

And for the second question, nope.. currently it does not predict if the variable defined in the script. Script's result could depend on actions like `sendRequest`, so there seem no simple way to quickly predict it.

**Comment by subnetmarco** [comment, marked as answer]
**Habemus pre-request scripting!** 

The new version of Insomnia 9.0 has been shipped with pre-request scripting available in the collections [among many other features](https://konghq.com/blog/product-releases/insomnia-9-0-pre-request-scripting-api-mocking). 

We will now work towards [enabling post-request scripting](https://github.com/Kong/insomnia/discussions/6242) which should ship in approximately one month, in the meantime let us know if there are any improvements you would like to see in pre-request scripting and we will address them in the next releases.

Known gaps that will be solved shortly in the next releases are:

* The `globals` environment variables SDK is not supported because we need to introduce first the concept of global environment variables in the product, which we also will ship shortly.
* The support for iteration data, we need to introduce support for selecting a local file and then make it available in the scripting SDK.

I will close this discussion for now.

**Comment by savoygrizzly** [comment]
Awesome ! Will it be possible to use some sort of basic require/import to at least have some default libs built-in ? I tried using Crypto with the same exact use case as the one that started this thread, couldn't import it or use it. Postman provides the deprecated CryptoJS by default I think. However having the latest standard one would be best I guess.

**Reply by ihexxa** [reply]
Hi @savoygrizzly, thanks for the feedback. Actually crypto-js should be part of the built-in libs but there was an issue in importing it, fixed in #7315 and will be included in the next minor release 9.1.

**Comment by ShakedAharonn** [comment]
Hi,
Thank you for adding support for the pre-request script!
Is there support for top level pre request script?
Or do you need to define it for every request?

my use-case - generating 1 time token from a const secret. this token is required for each api (i have about 30)

**Reply by subnetmarco** [reply]
We will support folder-level scripting in the next minor release, and afterwards collection-level scripting as well.

**Reply by ShakedAharonn** [reply]
Thank you, any estimation on when this will happen?

**Reply by subnetmarco** [reply]
@ShakedAharonn it should already be available in the latest 9.3 beta. We are planning a GA sometime next week.

**Reply by ShakedAharonn** [reply]
hi @subnetmarco 
Seems like 9.3 is officially out, but when i try to use folder level scripting it doesn't seem to work. couldn't find anything in the release doc or documentation. Do you know the status? should it work on 9.3?

**Comment by subnetmarco** [comment]
I wanted to let you know that we have also completed the beta for post-request scripting (or after response scripting). More info at https://github.com/Kong/insomnia/discussions/6242#discussioncomment-9533769
